# C.1 BOPS
Resuelve el siguiente caso :

1. ¿Deberían expandirse a Canadá?

2. ¿Cuántos millones de dólares se ganaron o perdieron a partir del programa? Explica
tu razonamiento y metodología.

Pista: Existen dos experimentos naturales. Canadá y las tiendas que se encuentran
lejos. Utilízalos.

## Respuestas

### 1)
Parece que sí pues BOPS tuvo un efecto positivo en general. Sin embargo, no es posible contestar a esta pregunta pues no tenemos información sobre las ventas en línea en Canadá.

### 2)
Estimamos que gracias a BOPS se ganaron 3 millones de dolares en 6 meses.

## Razonamiento

Según el texto del caso, tanto las ventas en linea como en tienda bajaron al añadir el sistema BOPS. Sin embargo, desde el punto de vista de los compradores, el sistema BOPS no quita ninguna opción de compra, solo se agrega una nueva forma de comprar, y por esto no es razonable suponer que el BOPS haya tenido un efecto negativo en las ventas. 

Es importante notar que el sistema BOPS solo se agregó en Estados Unidos y no en Canada. Así que para medir el efecto que ha tenido el BOPS en tiendas deberiamos considerar las ventas en ambos países por separado. Además, para medir el efecto que ha tenido el BOPS en las compras en linea podemos dividir la información entre compras desde zonas cercanas a una tienda y compras en zonas que no están cerca de una tienda.

Empezaremos sacando información general que nos servirá para calcular números como promedio de ventas por semana antes y después del BOPS.

In [76]:
import pandas as pd

In [77]:
sales_online = pd.read_csv('/Users/sebastianquintanilla/Downloads/BOPS_case/bops_online.csv')
sales_bm = pd.read_csv('/Users/sebastianquintanilla/Downloads/BOPS_case/bops_bm.csv')

#quitamos columnas que no tienen info
sales_bm.drop(['Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10'],inplace=True,axis=1)

In [78]:
#empezamos con los números para tiendas
n_stores = sales_bm['id (store)'].nunique()
n_usa = sales_bm[sales_bm['usa'] == 1]['id (store)'].nunique()
n_can = sales_bm[sales_bm['usa'] == 0]['id (store)'].nunique()

print('tiendas en USA: '+str(n_usa))
print('tiendas en Canada: '+str(n_can))
print('total de tiendas: '+str(n_stores))

tiendas en USA: 67
tiendas en Canada: 17
total de tiendas: 84


In [80]:
#ahora vamos con los numeros para ventas online
n_zones = sales_online['id (DMA)'].nunique()
n_close = sales_online[sales_online['close'] == 1]['id (DMA)'].nunique()
n_far = sales_online[sales_online['close'] == 0]['id (DMA)'].nunique()

print('Zonas cercanas a tiendas consideradas: '+ str(n_close))
print('Zonas lejanas a tiendas consideradas: '+ str(n_far))
print('Total de zonas consideradas: '+ str(n_zones))

Zonas cercanas a tiendas consideradas: 101
Zonas lejanas a tiendas consideradas: 109
Total de zonas consideradas: 210


In [102]:
#por último, calculamos cuantas semanas de la base son antes y despues del BOPS
w_bm = sales_bm['week'].nunique()
w_ol = sales_online['week'].nunique()

w_before_bm = sales_bm[(sales_bm['after'] == 0)&(sales_bm['usa'] == 1)]['week'].nunique()
w_after_bm = sales_bm[(sales_bm['after'] == 1)&(sales_bm['usa'] == 1)]['week'].nunique()

w_before_ol = sales_online[sales_online['after'] == 0]['week'].nunique()
w_after_ol = sales_online[sales_online['after'] == 1]['week'].nunique()


print('No. de semanas antes de BOPS: '+ str(w_before_bm) + '|' + str(w_before_ol))
print('No. de semanas despues de BOPS: '+ str(w_after_bm) + '|' + str(w_after_ol))
print('Total de semanas: '+ str(w_bm) + '|' + str(w_ol))

No. de semanas antes de BOPS: 27|26
No. de semanas despues de BOPS: 27|25
Total de semanas: 53|51


In [103]:
avg_can_before = sales_bm[(sales_bm['usa'] == 0)&(sales_bm['after'] == 0)][' sales '].sum()/(n_can*w_before_bm)
avg_can_after = sales_bm[(sales_bm['usa'] == 0)&(sales_bm['after'] == 1)][' sales '].sum()/(n_can*w_after_bm)

avg_usa_before = sales_bm[(sales_bm['usa'] == 1)&(sales_bm['after'] == 0)][' sales '].sum()/(n_usa*w_before_bm)
avg_usa_after = sales_bm[(sales_bm['usa'] == 1)&(sales_bm['after'] == 1)][' sales '].sum()/(n_usa*w_after_bm)

print('Promedio de ventas por tienda por semana ($), antes y despues de BOPS')
print('Canada: '+ str(avg_can_before) + '|' + str(avg_can_after))
print('USA: '+ str(avg_usa_before) + '|' + str(avg_usa_after))


Promedio de ventas por tienda por semana ($), antes y despues de BOPS
Canada: 66862.2592593|56325.2396514
USA: 67844.4969596|61058.9325594


In [104]:
#calculamos la razón de cambio en ambos países
rate_can = (avg_can_after - avg_can_before)/avg_can_before
rate_usa = (avg_usa_after - avg_usa_before)/avg_usa_before

print('Proporción de cambio del promedio de ventas')
print('En Canada: ' + str(rate_can))
print('En USA: ' + str(rate_usa))

Proporción de cambio del promedio de ventas
En Canada: -0.15759293396
En USA: -0.100016430283


Vemos que el promedio de ventas por tienda por semana bajó en ambos países despues del BOPS. Sin embargo en Canada las ventas bajaron 15% mientras que en USA bajaron 10%.

Dado que en Canada no se implementó el BOPS podemos ver que esta baja en las ventas no se debe al BOPS y podemos suponer que se debe a algún factor externo (por ejemplo baja en popularidad de las tiendas). 

Vemos que en ambos países el promedio de ventas por tienda por semana es similar antes del BOPS (~67 mil dolares). Resulta razonable suponer que sin el BOPS las ventas en USA se hubieran visto afectadas de forma similar que en Canada pues ambas poblaciones tenían un comportamiento muy parecido.

Con estas dos suposiciones (un factor externo ocasionó la baja en ventas, y este factor afecta de igual manera a ambas poblaciones) podemos especular que el BOPS ayudó a amortiguar esta baja en ventas en USA. Veamos ahora como afectó el BOPS a las ventas en linea.

In [108]:
avg_far_before = sales_online[(sales_online['close'] == 0)&(sales_online['after'] == 0)][' sales '].sum()/(n_far*w_before_ol)
avg_far_after = sales_online[(sales_online['close'] == 0)&(sales_online['after'] == 1)][' sales '].sum()/(n_far*w_after_ol)

avg_close_before = sales_online[(sales_online['close'] == 1)&(sales_online['after'] == 0)][' sales '].sum()/(n_close*w_before_ol)
avg_close_after = sales_online[(sales_online['close'] == 1)&(sales_online['after'] == 1)][' sales '].sum()/(n_close*w_after_ol)

print('Promedio de ventas en linea por zona por semana ($), antes y despues de BOPS')
print('Zonas lejanas a tiendas: '+ str(avg_far_before) + '|' + str(avg_far_after))
print('Zonas cercanas a tiendas: '+ str(avg_close_before) + '|' + str(avg_close_after))

Promedio de ventas en linea por zona por semana ($), antes y despues de BOPS
Zonas lejanas a tiendas: 15659.1503176|13770.9911927
Zonas cercanas a tiendas: 13743.5575019|11615.4590099


In [111]:
#calculamos la razón de cambio en ambas zonas
rate_far = (avg_far_after - avg_far_before)/avg_far_before
rate_close = (avg_close_after - avg_close_before)/avg_close_before

print('Proporción de cambio del promedio de ventas en linea')
print('Zonas lejanas a tiendas: ' + str(rate_far))
print('Zonas cercanas a tiendas: ' + str(rate_close))

Proporción de cambio del promedio de ventas en linea
Zonas lejanas a tiendas: -0.120578644857
Zonas cercanas a tiendas: -0.154843350545


Vemos que tanto en zonas cercanas a tiendas como en zonas lejanas, la cantidad de ventas bajó. De nuevo, las ventas en zonas lejanas a tiendas no deberían ser afectadas por el BOPS pues en zonas lejanas no tiene sentido usar el BOPS. Esto es consistente con el supuesto de un factor externo que afecta las ventas de la tienda en general.

Otra cosa razonable que vemos, es que en promedio se tengan mayores ventas en zonas lejanas, pues las compras en linea se vuelven una necesidad para los clientes que viven en estas zonas, mientras que los clientes que viven cerca a una tienda pueden decidir comprar en linea o ir a la tienda.

Vemos también que en zonas cercanas a las tiendas la baja en ventas es mayor, pues en las zonas lejanas se tiene una baja del 12% mientras que en zonas cercanas esta baja es del 15.5%.
Por lo tanto podemos suponer que la baja en ventas en linea se vio acentuada por la presencia del BOPS.

## conclusiones

Después de comparar las poblaciones que fueron afectadas por el BOPS (clientes en tiendas en USA y clientes en linea de zonas cercanas), contra las poblaciones para la cuales este nuevo método de compra no hizo ningún cambio (clientes en tiendas en Canada y clientes en linea en zonas lejanas), podemos hacer tres suposiciones razonables:
- La baja en ventas se debe a un factor externo. No podemos deducir este factor con la información que tenemos
- El BOPS tuvo un efecto negativo en las ventas en linea.
- El BOPS tuvo un efecto positivo en las ventas en tienda.

Dado que el BOPS tuvo efectos distintos en sus dos tipos de ventas, para poder responder a las preguntas debemos hacer una proyección de lo que hubiera pasado si no se hubiera implementado este nuevo método de compra.
Para la proyección lo que haremos es suponer que todas las ventas en tienda bajan por el mismo porcentaje que las ventas en tienda en Canada, y que todas las ventas en linea bajan por el mismo porcentaje que las ventas en linea en zonas lejanas.

In [117]:
sales_bm_before = sales_bm[sales_bm['after'] == 0][' sales '].sum()
sales_bm_after = sales_bm[sales_bm['after'] == 1][' sales '].sum()

sales_ol_before = sales_online[sales_online['after'] == 0][' sales '].sum()
sales_ol_after = sales_online[sales_online['after'] == 1][' sales '].sum()

proj_bm = sales_bm_before * (1+rate_can)
proj_ol = sales_ol_before * (1+rate_far)

print('Ventas en tienda antes de BOPS | Después de BOPS | Proyección sin BOPS')
print(str(sales_bm_before) + '|' + str (sales_bm_after) + '|' + str(proj_bm))

print('Ventas en linea antes de BOPS | Después de BOPS | Proyección sin BOPS')
print(str(sales_ol_before) + '|' + str (sales_ol_after) + '|' + str(proj_ol))

Ventas en tienda antes de BOPS | Después de BOPS | Proyección sin BOPS
153420472.0|136308894.0|129242489.688
Ventas en linea antes de BOPS | Después de BOPS | Proyección sin BOPS
80468614.0|66854985.0|70765817.5703


Para calcular el efecto total de BOPS sumamos la ventas despues de BOPS por ambos medios y restamos la suma de ambas proyecciones.

In [119]:
BOPS_effect = (sales_bm_after + sales_ol_after) - (proj_bm + proj_ol)
print('Efecto de BOPS: ' +str(BOPS_effect))

Efecto de BOPS: 3155571.74164


Bajo nuestras suposiciones, BOPS tuvo un efecto positivo de más de 3 millones de dólares en estos 6 meses. Aunque en general las ventas bajaron, sin este nuevo programa las ventas hubieran bajado aún más.

Parece que una expansión del programa a Canadá podría beneficiar a la empresa. Sin embargo nos falta información sobre las ventas en línea en Canadá, y cuántas de estas se hacen desde zonas lejanas a tiendas. 

Hay que tomar en cuenta que en Canadá la presencia en tiendas es pequeña (17 tiendas en Canadá contra 67 tiendas en USA). Por ello parece que el efecto de BOPS sería bajo, aunque no podemos saberlo con certeza.